src/main/java/com/f1soft/bachaat/service/impl/UserServiceImpl.java

```diff
@@ -30,6 +40,22 @@
 
     private static Logger logger = Logger.getLogger(UserServiceImpl.class.getName());
 
+    @Override
+    public UserResponseDTO addUser(UserRequestDTO userRequestDTO) {
+        logger.info("Inside Add User Service");
+        User user = userRepository.findByMobileNumber(userRequestDTO.getMobileNumber());
+        if (user != null) {
+            throw new UserAlreadyExistsException("User with " + user.getMobileNumber() + " mobile number already exist.");
```

Comment:
you can use String.format to wrap user's get mobile number method here.

Reply:
done